packages/node-sdk/batcher/src/batcher-server.ts

```diff
@@ -0,0 +1,291 @@
+            batcherAddress: Type.String(),
+            chainName: Type.String(),
+            batchIntervalSeconds: Type.Number(),
+            paimaL2Fee: Type.String(),
+            namespace: Type.String(),
+            maxBatchSize: Type.Number(),
+          }),
+          timestamp: Type.String(),
+        }),
+      },
+    },
+  }, async () => {
+    const stats = await run(() => batcher.getQueueStats());
+    const config = batcher.getPublicConfig();
+    return {
+      batcher: {
+        isRunning: stats.isRunning,
+        isProcessingBatch: stats.isProcessingBatch,
+        pendingInputs: stats.pendingInputs,
+      },
+      config,
+      timestamp: new Date().toISOString(),
+    };
+  });
+
+  // Get queue statistics
+  server.get("/queue-stats", {
+    schema: {
+      tags: ["developer"],
+      response: {
+        200: Type.Object({
+          pendingInputs: Type.Number(),
+          processingBatch: Type.Boolean(),
+          lastBatchTimestamp: Type.String(),
+        }),
+      },
+    },
+  }, async () => {
+    return await batcher.getQueueStats();
+  });
+
+  // Add user input to batcher
+  server.post("/send-input", {
+    schema: {
+      tags: ["batcher"],
+      body: BatchedSubunitSchema,
+      response: {
+        200: Type.Object({
+          success: Type.Boolean(),
+          message: Type.String(),
+          queueSize: Type.Number(),
+        }),
+      },
+    },
+  }, async (
+    request: FastifyRequest<{ Body: BatchedSubunit }>,
+    reply,
+  ) => {
+    try {
+      const batchedSubunit = request.body;
+
+      const success = await run(() => batcher.addUserInput(batchedSubunit));
+
+      if (success) {
+        const stats = await run(() => batcher.getQueueStats());
+        return {
+          success: true,
+          message: "Input added to batch queue",
+          queueSize: stats.pendingInputs,
+        };
+      } else {
+        return reply.status(400).send({
+          error: "Failed to add input to batch queue",
+        });
+      }
+    } catch (error) {
+      console.error("Error adding input to batcher:", error);
+      return reply.status(500).send({
+        error: "Internal server error",
+        message: error instanceof Error ? error.message : "Unknown error",
+      });
+    }
+  });
+
+  // TODO Only in dev mode.
+  // Force process current batch (useful for testing/debugging)
+  server.post("/force-batch", async (_, reply) => {
+    try {
+      await batcher.forceBatch();
+      const stats = await run(() => batcher.getQueueStats());
+      return {
+        success: true,
+        message: "Batch processing forced",
+        remainingInputs: stats.pendingInputs,
+      };
+    } catch (error) {
+      console.error("Error forcing batch:", error);
+      return reply.status(500).send({
+        error: "Failed to force batch processing",
+        message: error instanceof Error ? error.message : "Unknown error",
+      });
+    }
+  });
+
+  // TODO Only in dev mode.
```

Comment:
we should handle this, because if we forget this in a single production app it could go very bad

Reply:
PAI-28